pkg/controller/controller_instance.go

```diff
@@ -1342,11 +1342,13 @@ func (c *controller) prepareRequestHelper(instance *v1beta1.ServiceInstance, ser
 
 	// osb client handles whether or not to really send this based
 	// on the version of the client.
+	c.clusterIDLock.RLock()
 	rh.requestContext = map[string]interface{}{
-		"platform":  ContextProfilePlatformKubernetes,
-		"namespace": instance.Namespace,
+		"platform":   ContextProfilePlatformKubernetes,
+		"namespace":  instance.Namespace,
+		"cluster-id": c.clusterID,
```

Comment:
That is a super question. 
Reading through the spec, I don't see any definition of continuity there. 
https://github.com/openservicebrokerapi/servicebroker/blob/v2.13/profile.md

It does seem 'obvious' that it shouldn't change among different requests, but I'm not sure that we're technically restricted in that regard. @duglin WDYT? Need a spec addition or is it located somewhere else?

Reply:
The cluster ID should probably never change and if it does its going to be very very rare since that's the unique identifier for the cluster. You would normally only ever change it if something really serious happened to your entire state, like you need to reset it. So this is very much an edge edge case because even so odds are you wouldn't have anything provisioned yet either.

But... if it does happen I think Kube needs to send the very latest clusterID each time since that's the only valid value that the platform will understand. Typically the clusterID will be used when the broker talks back to the platform and it needs to tell it which cluster its talking about. If the ID is old then things won't work - so we need to provide the broker the latest version of the clusterID as known by Kube.